karapace/schema_reader.py

```diff
@@ -524,9 +524,16 @@ def get_referenced_by(self, subject: Subject, version: ResolvedVersion) -> Optio
         return self.database.get_referenced_by(subject, version)
 
     def _resolve_reference(self, reference: Reference) -> Dependency:
-        subject_data = self.database.find_subject_schemas(subject=reference.subject, include_deleted=False)
+        subject_data: Dict[ResolvedVersion, SchemaVersion] = self.database.find_subject_schemas(
```

Comment:
thought: This explicit annotation shouldn't be needed, since it's the exact return type of the called method.